source/macos/AdaptiveCards/AdaptiveCards/Renderers/FactSetRenderer.swift

```diff
@@ -0,0 +1,96 @@
+import AdaptiveCards_bridge
+import AppKit
+
+class FactSetRenderer: NSObject, BaseCardElementRendererProtocol {
+    static let shared = FactSetRenderer()
+    
+    func render(element: ACSBaseCardElement, with hostConfig: ACSHostConfig, style: ACSContainerStyle, rootView: NSView, parentView: NSView, inputs: [BaseInputHandler]) -> NSView {
+        guard let factSet = element as? ACSFactSet else {
+            logError("Element is not of type ACSFactSet")
+            return NSView()
+        }
+        let factArray = factSet.getFacts()
+        let factsetConfig = hostConfig.getFactSet()
+        
+        // Create horizontal Stack to add both views
+        let horizontalStack = ACRFactSetStackView()
+        horizontalStack.orientation = NSUserInterfaceLayoutOrientation.horizontal
+        horizontalStack.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Create stacks to hold title and value of FactSet
+        let titleStack = NSStackView()
+        let valueStack = NSStackView()
+        titleStack.orientation = .vertical
+        valueStack.orientation = .vertical
+        titleStack.translatesAutoresizingMaskIntoConstraints = false
+        valueStack.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Main loop to iterate over Array of facts
+        for fact in factArray {
+            let titleView = TitleFact()
+            let valueView = ACRFactTextField()
+            titleView.titleText.setLabel(string: fact.getTitle())
+            valueView.setLabel(string: fact.getValue())
+            
+            if let colorHex = hostConfig.getForegroundColor(style, color: .default, isSubtle: false), let textColor = ColorUtils.color(from: colorHex) {
+                titleView.titleText.labelText.textColor = textColor
+                valueView.labelText.textColor = textColor
+            }
+            
+            if !(titleView.titleText.labelText.stringValue.isEmpty) || !(valueView.labelText.stringValue.isEmpty) {
+                titleStack.addArrangedSubview(titleView.titleText)
+                valueStack.addArrangedSubview(valueView)
```

Comment:
I see it's the `titleText` inside the customView accessed and not the customView. The `TitleFact` class need not exist.